File: EZPZproject/apiEZPZ/src/main/java/com/example/apiezpz/post/repository/PostRepository.java
package com.example.apiezpz.post.repository;

import com.example.apiezpz.post.domain.Post;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;

@Repository
public interface PostRepository extends JpaRepository<Post, Long> {
    List<Post> findAllByOrderByIdDesc();  // ID 기준 내림차순 정렬
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/post/repository/PostRepository.kt
package com.example.apiezpz.post.repository

import com.example.apiezpz.post.domain.Post
import jakarta.transaction.Transactional
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PostRepository : JpaRepository<Post, Long> {
    fun findAllByOrderByIdDesc(): List<Post>  // ID 기준 내림차순 정렬
    
    @Transactional
    fun deleteByWriter(writer: String)
} 